### PR TITLE
Further gallery and doc tidying

### PR DESCRIPTION
**Pull request by Cadair**
This changes the output directory of the gallery to `generated/gallery` it also changes the `api` directory which is generated by `automodapi` to be `generated/api` the idea here is that it makes it clearer what content is being generated by sphinx and what is part of the repo.

Note: I chose `generated` not `_generated` because it makes the urls nicer.

This inspired by / copied from astropy/astropy#4924

### CODE REVIEW

**Comment by Cadair** [comment]
oh, it also makes sphinx-gallery optional for the doc build to run.

File: doc/source/conf.py [hunk 1]
```diff
@@ -1,34 +1,17 @@
...
-# import occurs, it will import the *source* version of astropy instead of the
-# version installed (if invoked as "make html" or directly with sphinx), or the
-# version in the build directory (if "python setup.py build_sphinx" is used).
-# Thus, any C-extensions that are needed to build the documentation will *not*
-# be accessible, and the documentation will not build correctly.
-
-# -- Mock Modules -------------------------------------------------------------
-# scikit-image will not install on RTD, so if it is no present we
-# use Mock instead so the docs will still build correctly.
-
```
Comment (dpshelio) [inline doc/source/conf.py]: so we don't need to mock anymore?

**Comment by dpshelio** [comment]
:+1:

File: doc/source/conf.py (hunk 1 above)
Comment (Cadair) [inline doc/source/conf.py]: nope, not since RTD supported conda

**Pull request merged**